### DISCUSSION TITLE
[Feature Request] Import Files as Individual Notes

### DISCUSSION THREAD

**Discussion by arcanecode** [standardnotes/forum#2541] [Suggestions] [answered]
Here's an idea for people who might wish to bulk upload a bunch of files into Standard Notes.

In the menu, user picks "Import, Folder".

They are prompted for the name of the folder, and optionally a Tag to use. If they don't supply a tag name, the name of the folder is used for the tag.

Every text based file in the selected file becomes its own note, with the name of the note being the file name, and using the tag the user indicated in the previous step.

If a note already exists with that name, user will be prompted to overwrite, overwrite all (i.e. don't ask again) or import with a new name. You could mimic the style of windows, add a (1) or a (2) etc, at the end of the note name.

For added functionality, have an option to also import all subfolders. Each note would then have the tag for the base folder, and additional tags for the subfolder(s) it is in. 

I can see this being used for people who want to backup their source code in a safe, encrypted spot. Also for people who have created a bunch of text files over time and want a place to easily organize them.

**Comment by atmoio** [comment]
We do have this functionality with the plaintext converter tool: https://dashboard.standardnotes.org/tools

We don't like to couple this functionality into the core app as to keep concerns separate. But in the future I do want to add a tag option to the converter script so it adds the notes to a desired tag.

**Comment by myreli** [comment, marked as answer]
There is now an importer tool built-in on Standard Notes: https://standardnotes.com/help/11/how-can-i-import-notes-from-a-bunch-of-text-files. 

<img width="640" alt="image" src="https://user-images.githubusercontent.com/17554234/218776457-9966abb2-596d-4084-97df-593c4c7916c7.png">
